src/widgets/splits/SplitInput.cpp

```diff
@@ -256,15 +257,12 @@ void SplitInput::themeChangedEvent()
 
 void SplitInput::updateEmoteButton()
 {
-    float scale = this->scale();
-
-    QString text = "<img src=':/buttons/emote.svg' width='xD' height='xD' />";
-    text.replace("xD", QString::number(int(12 * scale)));
+    auto scale = this->scale();
 
-    if (this->theme->isLightTheme())
-    {
-        text.replace("emote", "emoteDark");
-    }
+    auto text =
+        QStringLiteral("<img src=':/buttons/%1.svg' width='%2' height='%2' />")
+            .arg(this->theme->isLightTheme() ? "emoteDark" : "emote")
+            .arg(int(12 * scale));
```

Comment:
Not simply, no. Overload selection becomes spaghetti. (the width/height param becomes the field width instead).
This could be "fixed" either by pre-massaging the param or by changing param order but that feels off.

Reply:
Yeah, creating a temporary QString isn't super nice (hence the comment on multi-arg). [`QStringBuilder`](https://doc.qt.io/qt-6/qstring.html#more-efficient-string-construction) could also be used.

Reply:
If it was just one concat that'd be fair, but with 3 replacements it'd look kinda bad with QStringBuilder
```c++
auto text = "<img src=':/buttons/" << this->theme->isLightTheme() ? "emoteDark" : "emote" << "' width='" << int(12 * scale) << "' height='" << int(12 * scale) << "' />";
```

I might not understand what multi-arg is, I assumed it was [this](https://doc.qt.io/qt-6/qstring.html#arg-14)

Both options seem severely suboptimal to me

Reply:
Yea, that's multi-arg, I was just listing alternatives. Note that QStringBuilder would use `%` to concatenate.